caffe2/operators/generate_proposals_op.cc

```diff
@@ -171,17 +171,20 @@ void GenerateProposalsOp<CPUContext>::ProposalsForOneImage(
   // 3. remove predicted boxes with either height or width < min_size
   auto keep = utils::filter_boxes(proposals, min_size, im_info);
   DCHECK_LE(keep.size(), scores.size());
-
-  // 4. sort all (proposal, score) pairs by score from highest to lowest
-  // 5. take top pre_nms_topN (e.g. 6000)
-  std::sort(keep.begin(), keep.end(), [&scores](int lhs, int rhs) {
```

Comment:
Maybe you can use `std::partial_sort` to replace `std::nth_element` + `std::sort`. 

http://en.cppreference.com/w/cpp/algorithm/partial_sort